llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/Input/InputBase.td

```diff
@@ -113,4 +108,21 @@ def IREEInput_TiedOpStorageAttr :
   let constBuilderCall = "$_builder.getI64ArrayAttr($0)";
 }
 
+//===----------------------------------------------------------------------===//
+// Type aliases for working with shapes
+//===----------------------------------------------------------------------===//
+
+def IREEInput_Dim : TypeAlias<Index>;
+def IREEInput_Dims : Variadic<IREEInput_Dim>;
+def IREEInput_Shape : Variadic<IREEInput_Dim>;
```

Comment:
Is there anything that could be added here to document the differences? E.g., Dims and Shape seems the same and it is not clear which one should be used when.

Reply:
I'm not familiar with the original typedefs to write anything meaningful here, will get to it later together with declarative assembly cleanup